src/Samples/dotnet6server/EchoService.cs

```diff
@@ -0,0 +1,30 @@
+using System.Threading.Tasks;
+using MyContracts;
+using CoreWCF;
+
+namespace dotnet6Server
+{
+    public class EchoService : IEchoService
+    {
+        public string Echo(string text)
+        {
+            System.Console.WriteLine($"Received {text} from client!");
+            return text;
+        }
+
+        public string ComplexEcho(EchoMessage text)
+        {
+            System.Console.WriteLine($"Received {text.Text} from client!");
+            return text.Text;
+        }
+
+        public string FailEcho(string text)
+            => throw new FaultException<EchoFault>(new EchoFault() { Text = "WCF Fault OK" }, new FaultReason("FailReason"));
+
+        [AuthorizeRole("CoreWCFGroupAdmin")]
+        public string EchoForPermission(string echo)
+        {
```

Comment:
The EchoForPermission method with the AuthorizeRoleAttribute is demonstrating using Windows group membership to restrict the ability to call certain operations. This method is uncallable as you don't have anything configured to populate the Windows auth claimset. Can you remove the EchoForPermission method from the interface and implementation as it just confuses things.